sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_eventhandler.py

```diff
@@ -48,30 +47,28 @@
 
 
 # [START stream_event_handler]
-class MyEventHandler(AgentEventHandler):
+class MyEventHandler(AgentEventHandler[str]):
+
     def on_message_delta(self, delta: "MessageDeltaChunk") -> None:
-        for content_part in delta.delta.content:
-            if isinstance(content_part, MessageDeltaTextContent):
-                text_value = content_part.text.value if content_part.text else "No text"
-                print(f"Text delta received: {text_value}")
+        self.custom_data = f"Text delta received: {delta.text}"
```

Comment:
Why are we mutating the instance with the data? This seem like a very suspicious use of an instance attribute?

Reply:
Because when I try to use event handler, I discovered that it is very hard to pass content from the event handler function to where you trigger the streaming.   You certainly can pass an object to the event handler and change it's content in an event function, but you can't access the changes until the stream is completed.   So I introduced this custom_data.   Please take a look this sample:
https://github.com/Azure/azure-sdk-for-python/blob/3888296412e8fa88bdbae3540326b87efc4cd56e/sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_eventhandler.py#L87

Reply:
Using an instance variable to keep state is similar to using a global variable/not a good design pattern. If you want event handlers to extract information from individual events, it would be better to have them return whatever you wanted to give to the caller.

Reply:
Do you mean you prefer the sample to show?
`
class MyEventHandler(AgentEventHandler[str]):

    def on_message_delta(self, delta: "MessageDeltaChunk") -> Optional[str]:
        return f"Text delta received: {delta.text}"

    def on_thread_message(self, message: "ThreadMessage") -> Optional[str]:
         return f"ThreadMessage created. ID: {message.id}, Status: {message.status}"

    def on_thread_run(self, run: "ThreadRun") -> Optional[str]:
        return = f"ThreadRun status: {run.status}"

    def on_run_step(self, step: "RunStep") -> Optional[str]:
        return f"RunStep type: {step.type}, Status: {step.status}"

    def on_error(self, data: str) -> Optional[str]:
        return f"An error occurred. Data: {data}"

    def on_done(self) -> Optional[str]:
        return "Stream completed."

    def on_unhandled_event(self, event_type: str, event_data: Any) -> Optional[str]:
        return  f"Unhandled Event Type: {event_type}, Data: {event_data}"
`
with
`
for _, _, return_data in stream: 
   print(return_data)
`

Reply:
Pushed that in already.